extension/src/background/messageListener/popupMessageListener.ts

```diff
@@ -187,30 +230,59 @@ export const popupMessageListener = (request: Request) => {
 
   const confirmPassword = async () => {
     const { password } = request;
-    let keyStore;
-    try {
-      keyStore = await keyManager.loadKey(
-        localStorage.getItem(KEY_ID) || "",
-        password,
-      );
-    } catch (e) {
-      console.error(e);
-    }
-    let extra = { mnemonicPhrase: "" };
-    let publicKey = "";
-    let privateKey = "";
-
-    if (keyStore) {
-      ({ privateKey, publicKey, extra } = keyStore);
-      const { mnemonicPhrase } = extra;
-      store.dispatch(logIn({ publicKey, mnemonicPhrase }));
-      sessionTimer.startSession({ privateKey });
+    const keyIdList = JSON.parse(localStorage.getItem(KEY_ID_LIST) || `[]`);
+
+    // migration needed
+    if (!keyIdList.length) {
+      keyIdList.push(localStorage.getItem(KEY_ID));
+      localStorage.setItem(KEY_ID_LIST, JSON.stringify(keyIdList));
     }
 
+    const unlockedAccounts = [] as Array<string>;
+    let selectedPublicKey = "";
+    let selectedPrivateKey = "";
+    let accountMnemonicPhrase;
+
+    await Promise.all(
```

Comment:
because we don't story any public keys in plain text, to get all the accounts the user is working with, we need to iterate over all the key stores we have